sumcheck/src/util.rs

```diff
@@ -98,9 +103,31 @@ fn serial_batch_inversion_and_mul<F: PrimeField>(v: &mut [F], coeff: &F) {
 }
 
 pub(crate) fn extrapolate<F: PrimeField>(points: &[F], weights: &[F], evals: &[F], at: &F) -> F {
+    inner_extrapolate::<F, true>(points, weights, evals, at)
+}
+
+pub(crate) fn serial_extrapolate<F: PrimeField>(
+    points: &[F],
+    weights: &[F],
+    evals: &[F],
+    at: &F,
+) -> F {
+    inner_extrapolate::<F, false>(points, weights, evals, at)
+}
+
+fn inner_extrapolate<F: PrimeField, const IS_PARALLEL: bool>(
+    points: &[F],
+    weights: &[F],
+    evals: &[F],
+    at: &F,
+) -> F {
     let (coeffs, sum_inv) = {
         let mut coeffs = points.iter().map(|point| *at - point).collect::<Vec<_>>();
-        batch_inversion(&mut coeffs);
+        if IS_PARALLEL {
+            batch_inversion(&mut coeffs);
```

Comment:
Have we actually ever run any benchmarks to see whether parallel batch inversion helps us?

Reply:
Update: I just rigged up some benchmarks, and we do get a speedup here.  At least when doing a (batch) single inversion at a time (of eg `1<<20` field elements.)